### ISSUE TITLE
kubelet 1.13 does not configure Node status.addresses[].Hostname

### ISSUE THREAD

**Issue by mhulscher** [kubernetes/kubernetes#72237]
**What happened**:
When starting kubelet 1.13.1 the resulting Node object does not contain a Hostname as a valid address. This leaves the InternalIP as the only valid address. We run Kubernetes in Kubernetes and the InternalIP is not routable from the kube-apiserver, we rely on the hostname. Without the hostname, there is no way for the kube-apiserver to connect to the kubelet.

**What you expected to happen**:
Kubelet should register itself with Hostname.

Example of how it looks in 1.11:
```
$ kubectl get no node1 -o json | jq .status.addresses
[
  {
    "address": "10.100.0.249",
    "type": "InternalIP"
  },
  {
    "address": "node1",
    "type": "Hostname"
  }
]
```

Example of how it looks in 1.13:
```
$ kubectl get no node1 -o json | jq .status.addresses
[
  {
    "address": "10.100.0.249",
    "type": "InternalIP"
  }
]
```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
Tested a kubelet both with and without `--hostname-override`

**Environment**:
- Kubernetes version (use `kubectl version`): 1.13.1
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): `Container Linux by CoreOS 1800.7.0 (Rhyolite)`
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

**Comment by mhulscher** [comment]
@kubernetes/sig-node-bug

**Comment by liggitt** [comment]
what cloud provider are you using?

**Comment by mhulscher** [comment]
An external one implemented as a CCM. Is the CCM responsible for setting this field?

**Comment by liggitt** [comment]
yes. when `--cloud-provider=external` is set, the node does not set any address information itself.

https://github.com/kubernetes/kubernetes/blob/0c7e207904edfd2f934cbcf3167229fc9d491dca/pkg/kubelet/nodestatus/setters.go#L75-L84

that was true in 1.11 as well:

https://github.com/kubernetes/kubernetes/blob/b875d52ea96d2944468de43be21e9c059a662968/pkg/kubelet/kubelet_node_status.go#L456-L465

did you change to using the external cloud provider in 1.13?

**Comment by mhulscher** [comment]
Apologies. This should indeed be fixed in our CCM. Thanks for the feedback @liggitt . Closing this ticket.
